src/oop/special_class_method/hard/q7/MainTest.java

```diff
@@ -0,0 +1,251 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Java
+    private final String className = "PiggyBank";
+    private final String doubleFieldName = "money";
+    private final String intFieldName = "time";
+    private final String setIntMethodName = "setTime";
+    private final String getIntMethodName = "getTime";
+    private final String getDoubleMethodName = "getMoney";
+    private final String setDoubleMethodName = "setMoney";
+    private final String testClassName = "TestClass";
+    private final double outputMoney = 125.5;
+    private final int outputTime = 30;
+    private ObjectTest testClass;
+    private ObjectTest outputClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q7." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.special_class_method.hard.q7." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+    }
+
+    @Test
+    public void correctTestClass() throws Throwable {
+        Object classInstance = outputClass.createInstance();
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("I have " + outputMoney + " dollars in me after " + outputTime + " days."),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    public void piggyBankClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(doubleFieldName, double.class),
+                "Your " + className + " class is missing the " + doubleFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(doubleFieldName, "private"),
+                "Your " + doubleFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(intFieldName, int.class),
+                "Your " + className + " class is missing the " + intFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(intFieldName, "private"),
+                "Your " + intFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 300),
+                Arguments.of(234.4, 100),
+                Arguments.of(654.45, 365),
+                Arguments.of(43.3, 14)
+
+        );
+    }
+
```

Comment:
There aren't asserts to check if the constructors exist, only that they initialize attributes correctly and have the correct modifier